python/pyspark/pandas/window.py

```diff
@@ -706,10 +706,16 @@ def _apply_as_series_or_frame(self, func: Callable[[Column], Column]) -> FrameLi
         if groupby._agg_columns_selected:
             agg_columns = groupby._agg_columns
         else:
+            # pandas doesn't keep the groupkey as a column from 1.3 for DataFrameGroupBy
+            if isinstance(groupby, DataFrameGroupBy):
+                for groupkey in groupby._groupkeys:  # type: ignore
+                    groupby._column_labels_to_exclude.add(  # type: ignore
```

Comment:
Shall we not update `groupby._column_labels_to_exclude` directly?

```py
column_labels_to_exclude = groupby._column_labels_to_exclude.copy()
if isinstance(groupby, DataFrameGroupBy):
    for ...
        column_labels_to_exclude.add( ... )
agg_columns = [ ... if label not in column_labels_to_exclude]
```

Reply:
Oh, I got it. Let me address